### ISSUE TITLE
File Header Rule - Check Filename

### ISSUE THREAD

**Issue by aamctustwo** [realm/SwiftLint#1079]
### Summary

One of the more common errors that I see in the file header declaration is that the filename has been changed over time but the declaration comments at the top have not been updated to match. I don't believe this is currently possible to detect in the `file_header` rule, but would be a welcome addition.

### Example

The following would be a violation:

Filename: `Bar.swift`

File Header:
```swift
//
//  Foo.swift
//  Awesome App
//
//  Created by Aaron McTavish on 01/01/2016.
//  Copyright © 2016 Aaron McTavish. All rights reserved.
//
```

**Comment by jpsim** [comment]
Maybe as a "variable" replacement in the configuration? Shouldn't be too hard.

**Comment by marcelofabri** [comment]
That's what I thought, adding a placeholder to the regexes that would be replaced with the filename.

**Comment by nevil** [comment]
I did a hack (maybe not so user friendly) that allows to specify the capture group that should match the filename.
I did it this way to avoid re-creating the regexes on a per file basis.

Originally I thought I could use named capture groups as ICU supports it, but it seems this has not yet reached NSRegularExpression :-( . (http://userguide.icu-project.org/strings/regexp)

changes: 
https://github.com/realm/SwiftLint/compare/master...nevil:ah-file-header-filename-pattern

**Comment by nevil** [comment]
WIP Pull request created: https://github.com/apple/swift-corelibs-foundation/pull/1522

**Comment by marcelofabri** [comment]
Implemented on #2166 🚀
